api/common/sharing/validators.go

```diff
@@ -0,0 +1,56 @@
+package sharing
+
+import (
+	"context"
+	"github.com/Peripli/service-manager/pkg/instance_sharing"
+	"github.com/Peripli/service-manager/pkg/log"
+	"github.com/Peripli/service-manager/pkg/types"
+	"github.com/Peripli/service-manager/pkg/util"
+	"github.com/Peripli/service-manager/pkg/web"
+	"github.com/Peripli/service-manager/storage"
+	"github.com/tidwall/gjson"
+)
+
+func ValidateReferencedInstance(body []byte, tenantIdentifier string, repository storage.Repository, ctx context.Context, getTenantId func() string) (string, error) {
```

Comment:
is this a validator? i think the naming/location is wrong

Reply:
i mean this is not a validation - it returns the referenced instance ID from the request. the validation is done along the way - if it is not valid you cannot get the instance ID. I would name it "getReferencedInstanceID" or "extractReferenceInstanceID", not "valitate*".